### PR TITLE
fix(fx-2821): force re-render when pathname changes

### PR DESCRIPTION
**Pull request by dzucconi**
Closes: [FX-2821](https://artsyproduct.atlassian.net/browse/FX-2821)

### CODE REVIEW

File: src/v2/Apps/Collect/Routes/Collection/index.tsx [hunk 1]
```diff
@@ -114,6 +114,9 @@ export const CollectionApp: React.FC<CollectionAppProps> = props => {
             )}
             <Box>
               <ArtworkFilterContextProvider
+                // It's possible to navigate between collections directly.
+                // Force a re-render when the pathname changes.
+                key={location.pathname}
```
Comment (mzikherman) [inline src/v2/Apps/Collect/Routes/Collection/index.tsx:119]: cool, does this work for the collection pages? Maybe we should add this to the other instances? I'm pretty agnostic (front) on this, I just had that `routeTracking` type approach in mind is all.

Comment (dzucconi) [reply src/v2/Apps/Collect/Routes/Collection/index.tsx]: Eh, no you're right — pushed a more general solution.

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilterContext.tsx [hunk 2]
```diff
@@ -331,6 +332,22 @@ export const ArtworkFilterContextProvider: React.FC<
...
+  const { router, match } = useRouter()
+  const currentPathname = match?.location?.pathname
+
+  useEffect(() => {
+    if (!router || !currentPathname) return
+
+    router.addNavigationListener(({ pathname: nextPathname }) => {
+      if (currentPathname !== nextPathname) {
+        artworkFilterContext.resetFilters()
+      }
```
Comment (damassi) [inline src/v2/Components/v2/ArtworkFilter/ArtworkFilterContext.tsx:345]: wow, nice, not so bad 👍

Comment (mzikherman) [reply src/v2/Components/v2/ArtworkFilter/ArtworkFilterContext.tsx:345]: oh awesome! i didn't think of throwing a `useEffect` right in here.

**Comment by mzikherman** [comment]
Let's go!

**Pull request merged**